timemachine/cpp/src/kernels/k_neighborlist.cuh

```diff
@@ -201,11 +211,9 @@ void __global__ k_find_blocks_with_ixns(
 
     int neighborsInBuffer = 0;
 
-    const unsigned int atom_i_idx = blockIdx.x * blockDim.x + threadIdx.x;
-
-    RealType pos_i_x = atom_i_idx < NR ? row_coords[atom_i_idx * 3 + 0] : 0;
-    RealType pos_i_y = atom_i_idx < NR ? row_coords[atom_i_idx * 3 + 1] : 0;
-    RealType pos_i_z = atom_i_idx < NR ? row_coords[atom_i_idx * 3 + 2] : 0;
+    RealType pos_i_x = atom_i_idx < N ? coords[atom_i_idx * 3 + 0] : 0;
+    RealType pos_i_y = atom_i_idx < N ? coords[atom_i_idx * 3 + 1] : 0;
+    RealType pos_i_z = atom_i_idx < N ? coords[atom_i_idx * 3 + 2] : 0;
```

Comment:
We do this in a lot of places and seems to be our convention. No real preference other than to have a consistent way of doing it.

Reply:
In this case it seems like the redundant checks could actually have a performance impact, since this is "hot" kernel code. I guess they are likely optimized away, though?

Reply:
1) formatting is a litlte cleaner with ternary
2) ternary statements are always faster than branching ifs
3) if statements *may* be optimized into a ternary, but it's not guaranteed.

Reply:
4. btw: the initial loading of the coordinates isn't really a bottleneck